library/src/main/java/com/novoda/downloadmanager/lib/DownloadInfo.java

```diff
@@ -480,7 +480,11 @@ public void updateStatus(int status) {
     }
 
     private boolean isClientReadyToDownload(DownloadBatch downloadBatch) {
-        return downloadClientReadyChecker.isAllowedToDownload(downloadBatch);
+        return downloadClientReadyChecker.isAllowedToDownload(marshallToDownload(downloadBatch));
+    }
+
+    private Download marshallToDownload(DownloadBatch downloadBatch) {
+        return new Download(downloadBatch.getBatchId(), downloadBatch.getCurrentSize(), downloadBatch.getTotalSize());
```

Comment:
Ok, in that case `DownloadInfo` should probably be renamed to `FileDownloadInfo` or something similar.